test/integration/kubernetes_deploy_test.rb

```diff
@@ -806,6 +806,46 @@ def test_invalid_context
     end
   end
 
+  def test_multiple_configuration_files
+    old_config = ENV['KUBECONFIG']
+    begin
+      config_file = File.join(__dir__, '../fixtures/kube-config/unknown_config.yml')
+      ENV['KUBECONFIG'] = config_file
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
+        'Result: FAILURE',
+        'Configuration invalid',
+        "Kube config not found at #{config_file}"
+      ], in_order: true)
+
+      ENV['KUBECONFIG'] = " : "
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
+        'Result: FAILURE',
+        'Configuration invalid',
+        "Kube config file name(s) not set in $KUBECONFIG"
+      ], in_order: true)
+
+      ENV['KUBECONFIG'] = nil
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
+        'Result: FAILURE',
+        'Configuration invalid',
+        "$KUBECONFIG not set"
+      ], in_order: true)
+
+      valid_config = File.join(__dir__, '../fixtures/kube-config/valid_config.yml')
+      ENV['KUBECONFIG'] = "#{old_config}:#{valid_config}"
+      result = deploy_fixtures('hello-cloud')
```

Comment:
We should deploy only a tiny subset (suggestion: the configmap) to make the test faster, since the contents are actually irrelevant.

Reply:
Couldn't agree more. The test time is reduced from 14sec to 1.5sec after the suggested change.